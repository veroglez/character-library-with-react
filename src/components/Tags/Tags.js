import { arrayOf, func, string } from 'prop-types';
import React, { PureComponent } from 'react';
import scss from './Tags.module.scss';

class Tags extends PureComponent {
  static propTypes = {
    onFilter: func,
    tags: arrayOf(string).isRequired,
  };

  static defaultProps = {
    onFilter: undefined,
  }

  render() {
    const { props: { onFilter, tags } } = this;
    const service = 'character?species=';

    return (
      <div>
        {tags.map((tag, index) => (
          <div key={index.toString()} onClick={() => onFilter(`${service}${tag}`)}>{tag}</div> //eslint-disable-line
        ))}
      </div>
    );
  }
}

export default Tags;
