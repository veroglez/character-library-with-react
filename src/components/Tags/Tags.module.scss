@import "../../common/style.scss";

.container {
  padding: 0px 20px 40px;
  
  @include tablet {
    padding: 0px 60px 40px;
  }
}

.characters {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}