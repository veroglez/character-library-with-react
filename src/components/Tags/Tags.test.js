import React from 'react';
import renderer from 'react-test-renderer';

import Tags from './Tags';

const TAGS = ['alive', 'death'];
const FUNC = () => {};

describe('<Tags>', () => {
  it('renders when {tags}', () => {
    const tree = renderer.create(<Tags tags={TAGS} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('when {onFilter}', () => {
    const tree = renderer.create(<Tags tags={TAGS} onFilter={FUNC} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
