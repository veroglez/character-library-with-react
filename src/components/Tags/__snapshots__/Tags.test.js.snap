// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tags> renders when {tags} 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    <p
      className="undefined undefined"
    >
      alive
    </p>
  </div>
  <div
    onClick={[Function]}
  >
    <p
      className="undefined undefined"
    >
      death
    </p>
  </div>
</div>
`;

exports[`<Tags> when {onFilter} 1`] = `
<div>
  <div
    onClick={[Function]}
  >
    <p
      className="undefined undefined"
    >
      alive
    </p>
  </div>
  <div
    onClick={[Function]}
  >
    <p
      className="undefined undefined"
    >
      death
    </p>
  </div>
</div>
`;
